### ISSUE TITLE
[OIDC] Env variables require both ARM_OIDC_REQUEST_TOKEN and ARM_OIDC_TOKEN but both aren't documented.

### ISSUE THREAD

**Issue by MitchellGerdisch** [pulumi/pulumi-azure-native#2868]
### What happened?

Using basic `pulumi new azure-typescript` stack and an ESC env named "azure-oidc" based on the code below:

- If only `ARM_OIDC_REQUEST_TOKEN` is set then `esc env run  ORG/azure-oidc -- pulumi up --skip-preview` hangs on the resource group create.
- If only `ARM_OIDC_TOKEN` is set, then the `esc env run  ORG/azure-oidc -- pulumi up --skip-preview` throws an error about missing token.
- If BOTH are set, then everything works as expected. But the documentation (https://www.pulumi.com/registry/packages/azure-native/installation-configuration/#oidc-pulumi-provider-configuration) only references `ARM_OIDC_REQUEST_TOKEN`

```
values:
  azure:
    login:
      fn::open::azure-login:
        clientId: X
        tenantId: Y
        subscriptionId: Z
        oidc: true
  environmentVariables:
    ARM_USE_OIDC: 'true'
    ARM_CLIENT_ID: ${azure.login.clientId}
    ARM_TENANT_ID: ${azure.login.tenantId}
    #   ### ON azure-native 2.14.0 and 2.15.0
    #   ### COMMENT OUT OIDC_REQUEST; UNCOMMENT OIDC_TOKEN => no token found error
    #   ### Comment out OIDC_TOKEN; UNCOMMENT OIDC_REQUEST_TOKEN => resource group create hangs
    #   ### UNCOMMENT both => works
    ARM_OIDC_REQUEST_TOKEN: ${azure.login.oidc.token}
    ARM_OIDC_TOKEN: ${azure.login.oidc.token}
    ARM_SUBSCRIPTION_ID: ${azure.login.subscriptionId}
    ARM_OIDC_REQUEST_URL: https://api.pulumi.com/oidc
```

### Example

See above

### Output of `pulumi about`

CLI          
Version      3.90.1
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME          VERSION
azure-native  2.15.0
nodejs        unknown

Host     
OS       darwin
Version  13.5
Arch     x86_64

This project is written in nodejs: executable='/Users/mitch/.nvm/versions/node/v18.18.2/bin/node' version='v18.18.2'

Found no pending operations associated with demo/dev

Backend        
Name           pulumi.com

Dependencies:
NAME                  VERSION
@pulumi/pulumi        3.92.0
@types/node           16.18.60
@pulumi/azure-native  2.15.0

### Additional context

I believe this is related to https://github.com/pulumi/pulumi-azure-native/issues/2609

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by toriancrane** [comment]
Adding to this thread to second that defining all of those environment variables seems to do the trick.

**Comment by thomas11** [comment]
@MitchellGerdisch, @toriancrane I believe this issue is fixed by #2887. Do you have ready-to-test repro?

**Comment by toriancrane** [comment]
@thomas11 I commented out `ARM_OIDC_REQUEST_TOKEN` and `ARM_OIDC_REQUEST_URL` and just left `ARM_OIDC_TOKEN` before running pulumi up/destroy and it is working for me.
